WordPress/gutenberg
# Migrating `StyleBook` to use updated `Composite` implementation

## What?

This PR updates [`StyleBook` in `@wordpress/edit-site`](https://github.com/WordPress/gutenberg/blob/abab9fcfa1ef0ae99b7324e1bb75d2f57033c8b1/packages/edit-site/src/components/style-book/index.js) to use the updated `Composite` implementation from #54225, and adds additional semantic context around the example blocks.


## Why?

In #54225, an updated implementation of `Composite` was added to `@wordpress/components`. As per #55224, all consumers of `Composite` need to migrate from the old version to the new version.


## How?

 - Removes `__unstableComposite` imports from `@wordpress/components`
 - Adds private `Composite*` exports from `@wordpress/components`
 - Refactors `Examples`/`Example` within `StyleBook` to use updated `Composite` components
 - Additionally, adds semantics around `Composite` to make UX clearer for assistive technology users.


## Testing Instructions

Navigate to the Style Book overview, by selecting "Styles" from the "Design" menu, then opening the Style Book. The style book frame should still render as one actionable button. Clicking this should take you to the main story book view.

<p align="center">
  <img src="https://github.com/WordPress/gutenberg/assets/159848/fc6a6297-10ff-46a1-a33a-1baf038ac9d9" alt="Design menu in the appearance editor sidebar, with the 'Styles' option circled in red" width="40%">
  <img src="https://github.com/WordPress/gutenberg/assets/159848/c0deff27-ad33-48f8-940f-2c381573c03d" alt="Styles menu in the appearance editor sidebar, with the 'Style Book' access button circled in red" width="40%">
  <img src="https://github.com/WordPress/gutenberg/assets/159848/09634de1-7167-482b-88fc-016c4ac5c966" alt="The style book overview, circled in red" width="80%">
</p>

The style book can also be accessed from the the styles sidebar header in the site editor.

<p align="center">
  <img src="https://github.com/WordPress/gutenberg/assets/159848/963da2cf-199f-4470-97c1-9ee51bdd0a6d" alt="Part of the site editor header, with the 'Styles' toggle button circled in red" width="40%">
  <img src="https://github.com/WordPress/gutenberg/assets/159848/cb60384b-681f-47a9-be62-ebf548ddbba4" alt="The site editor styles sidebar header, with the 'Styles Book' access button circled in red" width="40%">
</p>

Each item within the Style Book should present as an actionable button, which should open the relevant section in the styles sidebar.

<p align="center">
  <img src="https://github.com/WordPress/gutenberg/assets/159848/63d2e16e-ec29-4e64-b34e-88ef94858589" alt="The style book view, with an individual block example circled in red" width="80%">
</p>

### Testing Instructions for Keyboard

With the main style book view open, the set of block examples should present as buttons within a grid. These should be navigable by tabbing to the grid, and using the arrow keys to move up and down.

<p align="center">
  <img src="https://github.com/WordPress/gutenberg/assets/159848/250f34a9-98ce-4d01-aa9e-c742baff8c12" alt="The style book view, with the Voiceover rotor showing a single table titled 'Examples of blocks in the Text category'" width="80%">
</p>
